api/src/main/java/org/openmrs/module/fhir/api/impl/DiagnosticReportServiceImpl.java

```diff
@@ -68,16 +93,31 @@ public void setDao(FHIRDAO dao) {
 	}
 
 	@Override
-	public DiagnosticReport getDiagnosticReport(String id) {
-		// Find Diagnostic Report (Encounter) in OpenMRS database
-		EncounterService encounterService = Context.getEncounterService();
-		Encounter omrsDiagnosticReport = encounterService.getEncounterByUuid(id);
-		// Get corresponding Handler
-		String handlerName = omrsDiagnosticReport.getEncounterType().getName();
+	public DiagnosticReport getDiagnosticReport(final String reportId) {
+		loadHandlerMap();
+		List<Order> orders = dao.getOrdersByAccessionNumber(reportId);
+		int orderId = 0;
+		if ((orders != null) && !orders.isEmpty()) {
+			orderId = orders.get(0).getOrderId();
```

Comment:
Is there a possibility of `orders` having more than one item?

Reply:
@dkayiwa: Yes, but that's out of scope for what we're currently trying to do. At a future point, we're probably going to have to look at adding a model for DiagnosticReport directly, since multiple orders can correspond to a single accession number and multiple diagnostic reports can also correspond to a single accession number, but this is something to be solved when we get the interface to create and update accession numbers working. For right now, we're just trying to get some sensible data returned.